mm2src/coins/sia/src/transaction.rs

```diff
@@ -0,0 +1,1587 @@
+#[serde(rename_all = "camelCase")]
+pub struct SiafundInputV2 {
+    pub parent: SiafundElement,
+    pub claim_address: Address,
+    pub satisfied_policy: SatisfiedPolicy,
+}
+
+impl Encodable for SiafundInputV2 {
+    fn encode(&self, encoder: &mut Encoder) {
+        self.parent.encode(encoder);
+        self.claim_address.encode(encoder);
+        self.satisfied_policy.encode(encoder);
+    }
+}
+
+// https://github.com/SiaFoundation/core/blob/6c19657baf738c6b730625288e9b5413f77aa659/types/types.go#L197-L198
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct SiacoinInputV1 {
+    pub parent_id: SiacoinOutputID,
+    pub unlock_condition: UnlockCondition,
+}
+
+impl Encodable for SiacoinInputV1 {
+    fn encode(&self, encoder: &mut Encoder) {
+        self.parent_id.encode(encoder);
+        self.unlock_condition.encode(encoder);
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct SiacoinInputV2 {
+    pub parent: SiacoinElement,
+    pub satisfied_policy: SatisfiedPolicy,
+}
+
+impl Encodable for SiacoinInputV2 {
+    fn encode(&self, encoder: &mut Encoder) {
+        self.parent.encode(encoder);
+        self.satisfied_policy.encode(encoder);
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct SiafundOutput {
+    pub value: u64,
+    pub address: Address,
+}
+
+// SiafundOutput remains the same data structure between V1 and V2 however the encoding changes
+#[derive(Clone, Debug)]
+pub enum SiafundOutputVersion<'a> {
+    V1(&'a SiafundOutput),
+    V2(&'a SiafundOutput),
+}
+
+impl<'a> Encodable for SiafundOutputVersion<'a> {
+    fn encode(&self, encoder: &mut Encoder) {
+        match self {
+            SiafundOutputVersion::V1(v1) => {
+                CurrencyVersion::V1(&Currency::from(v1.value)).encode(encoder);
+                v1.address.encode(encoder);
+            },
+            SiafundOutputVersion::V2(v2) => {
+                encoder.write_u64(v2.value);
+                v2.address.encode(encoder);
+            },
+        }
+    }
+}
+
+// SiacoinOutput remains the same data structure between V1 and V2 however the encoding changes
+#[derive(Clone, Debug)]
+pub enum SiacoinOutputVersion<'a> {
+    V1(&'a SiacoinOutput),
+    V2(&'a SiacoinOutput),
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct SiacoinOutput {
+    pub value: Currency,
+    pub address: Address,
+}
+
+impl<'a> Encodable for SiacoinOutputVersion<'a> {
+    fn encode(&self, encoder: &mut Encoder) {
+        match self {
+            SiacoinOutputVersion::V1(v1) => {
+                CurrencyVersion::V1(&v1.value).encode(encoder);
+                v1.address.encode(encoder);
+            },
+            SiacoinOutputVersion::V2(v2) => {
+                CurrencyVersion::V2(&v2.value).encode(encoder);
+                v2.address.encode(encoder);
+            },
+        }
+    }
+}
```

Comment:
please move the `Encodable` implementation below `SiacoinOutputVersion` declaration.